Google.Api.Gax.Grpc/BidirectionalStreamingBase.cs

```diff
@@ -113,14 +113,15 @@ public virtual Task WriteCompleteAsync()
         /// <summary>
         /// Async stream to read streaming responses, exposed as an async sequence.
         /// The default implementation will use <see cref="GrpcCall"/> to extract a response
-        /// stream, and adapt it to <see cref="IAsyncEnumerator{T}"/>, passing in the specified cancellation
-        /// token on each call to the gRPC stream.
+        /// stream, and adapt it to <see cref="AsyncResponseStream{T}"/>.
         /// </summary>
         /// <remarks>
         /// If this method is called more than once, all the returned enumerators will be enumerating over the
-        /// same underlying response stream, which may cause confusion.
+        /// same underlying response stream, which may cause confusion. Additionally, the sequence returned by
+        /// this method can only be iterated over a single time. Attempting to iterate more than once will cause
+        /// an <see cref="InvalidOperationException"/>.
         /// </remarks>
-        public virtual IAsyncEnumerator<TResponse> GetResponseStream(CancellationToken cancellationToken) =>
-            new ResponseStreamAdapter<TResponse>(GrpcCall.ResponseStream, cancellationToken);
+        public virtual AsyncResponseStream<TResponse> GetResponseStream() =>
```

Comment:
Maybe add an optional parameter CancellationToken and pass it on to the constructor, so the call to GetAsyncEnumerator can be skipped, I think it makes sense given that AsyncResponseStream in an IAsyncEnumerator itself?

Reply:
That means there'd be *three* places you could specify a cancellation token - I'd rather not do that. At the moment it's reasonably simple: if you want to specify the same cancellation token for everything, use it as an `IAsyncEnumerable` and pass the cancellation token into GetAsyncEnumerator. If you want to specify a potentially-different cancellation token on each call, just do that on MoveNextAsync.